lib/stagehand/staging/checklist.rb

```diff
@@ -12,14 +12,16 @@ def self.related_commit_ids(commit)
         related_entries(commit.entries).collect(&:commit_id).select(&:present?).uniq
       end
 
-      def self.related_entries(entries)
+      def self.related_entries(entries, relation_filter = nil)
         entries = Array.wrap(entries)
         related_entries = []
 
         entries_to_spider = Array.wrap(entries)
         while entries_to_spider.present?
-          contained_matching = CommitEntry.contained.matching(entries_to_spider)
-          matching_commit_entries = CommitEntry.where(:commit_id => contained_matching.select(:commit_id))
+          contained_matching = CommitEntry.contained.matching(entries_to_spider).to_a
+          contained_matching.select! {|entry| relation_filter.call(entry) } if relation_filter.present?
+
+          matching_commit_entries = CommitEntry.where(:commit_id => contained_matching.collect(&:commit_id).uniq)
```

Comment:
/sigh, too bad this can't be a relation. I wonder if there's a way we could keep it as a single query unless a `relation_filter` is used. Especially since we now have to load the entire model regardless of whether we're filtering or not.

How about
```ruby
contained_matching = contained_matching.where(id: contained_matching.select(&relation_filter)) if relation_filter
```
Then we avoid `to_a` on the relation, and `collect` down below.